docs/storage-abstraction.md

```diff
@@ -0,0 +1,714 @@
+# Storage abstraction
+
+Scalar DB is middleware on top of existing storage/database systems and provides a storage/database abstraction and a storage/database-agnostic ACID transaction manager on top of the abstraction.
+You usually use the ACID transaction manager but can also use the storage/database abstraction directly.
+With the abstraction, you can write code without being aware of the storage/database implementation.
+
+Note that the storage abstraction operations are not transactional, and data inconsistencies can happen when failures happen during the execution of the operations.
```

Comment:
```suggestion
Note that the storage abstraction operations are not transactional, and data inconsistencies can happen when failures occur during the execution of the operations.
```
To avoid the repetition of "happen"